### ISSUE TITLE
RSW: default `sssd` template is not valid

### ISSUE THREAD

**Issue by pat-s** [rstudio/helm#79]
With the default `sssd` dummy template, `sssd` complains during startup

chart `v0.5-rc4`

```
(Fri Nov  5 07:58:44:879305 2021) [sssd] [sss_ini_get_config] (0x0020): Config merge error: File /etc/sssd/conf.d/..data did not match provided patterns. Skipping.
(Fri Nov  5 07:58:44:879322 2021) [sssd] [sss_ini_get_config] (0x0020): Config merge error: File /etc/sssd/conf.d/mysssd did not match provided patterns. Skipping.
(Fri Nov  5 07:58:44:879325 2021) [sssd] [sss_ini_get_config] (0x0020): Config merge error: File /etc/sssd/conf.d/..2021_11_05_07_56_24.441839353 did not match provided patterns. Skipping.
(Fri Nov  5 07:58:44:879329 2021) [sssd] [sss_ini_get_config] (0x0020): Config merge error: Errors detected while parsing: /etc/sssd/conf.d/mysssd.conf.
(Fri Nov  5 07:58:44:879332 2021) [sssd] [sss_ini_get_config] (0x0020): Config merge error: Error (8) on line 1: Failed to read line.
(Fri Nov  5 07:58:44:879335 2021) [sssd] [sss_ini_get_config] (0x0020): Config merge error: Due to errors file /etc/sssd/conf.d/mysssd.conf is not considered. Skipping.
```

I am not fully sure if this really causes a startup abort.

One should usually always set this anyhow for automatic user provisioning, but maybe the default template could at least make `sssd` happy on a basic level :)

**Comment by colearendt** [comment]
Weeeird! Is this with a values file present? Do you mind sharing your values file? Or at least the `config.userProvisioning` part?

**Comment by pat-s** [comment]
Sure.

I was using

```yml
  userProvisioning:
    mysssd.conf:
      sssd:
        config_file_version: 2
        services: nss, pam
        domains: rstudio.com
      domain/rstudio.com:
        id_provider: ldap
        auth_provider: ldap
```

which is the example shown in the README.
But this seems also auto-populated when not defining it.

I did not expect this to work for our setup but just wanted to see how it maps in the pod.
At first I thought this triggers a startup failure of the `rstudio` container but maybe not - my current startup issue might have its root somewhere else.

**Comment by pat-s** [comment]
I think its fine. These messages

```(Wed Nov 17 21:03:18:784335 2021) [sssd] [sss_ini_get_config] (0x0020): Config merge error: File /etc/sssd/conf.d/..data did not match provided patterns. Skipping.
(Wed Nov 17 21:03:18:784374 2021) [sssd] [sss_ini_get_config] (0x0020): Config merge error: File /etc/sssd/conf.d/mysssd did not match provided patterns. Skipping.
(Wed Nov 17 21:03:18:784382 2021) [sssd] [sss_ini_get_config] (0x0020): Config merge error: File /etc/sssd/conf.d/..2021_11_17_21_03_15.972880182 did not match provided patterns. Skipping.
```

appear only because these are the mounts linking to the actual config files. They don't block startup actions.

**Comment by colearendt** [comment]
Sounds good! Let us know if you get stuck getting the workbench pod running! Feel free to send the logs and values file over to support@rstudio.com 😄  Hopefully that will be enough to determine the issue.

**Comment by pat-s** [comment]
Thanks.
Is that the preferred way to report feedback related to the helm chart instead of opening issues? Today I got told that I/we might be reporting issues the wrong way or out-of-scope questions, hence I am currently not sure what might be the best way to move forward. Also usually at support@rstudio.com people reply that are not necessarily engineers which adds some overhead when asking devops related Q's :)

I am still struggling to get `sssd` going as the users are not picked up even though `ldapsearch` works fine and the settings are identical to a config of another (docker) instance on which everything works fine. I've already spent days on this and it starts to get frustrating :/

Thanks for your help so far!

**Comment by colearendt** [comment]
I can definitely relate to that frustration - sorry for your troubles!! Why don't you reach out to your account rep and see if they can set up a call - feel free to mention me / this issue - they can pull me into the call. I think we'd be able to straighten some of this out pretty quickly, and can hopefully try to clarify some of the nuance as far as routing for you 😄  It can be really muddy sometimes 🙈  That said, your account rep is always a safe bet to help route questions if you're not sure.

Generally speaking, feedback on the helm chart is more than welcome here! (and your feedback has already improved the charts measurably! Thank you!). When it comes to environment configuration and getting sssd working in your particular environment, that's something we usually route through support and pull in the appropriate resources as needed 😄  We definitely want to get you unblocked and moving, so the account rep is probably the right point of contact at this point!
